### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by sublet**
Primarily just moving logic around so that it can better handle the direction we want to go in an organized fashion.  

Integrates options for async/await. 
Added a SRC folder where the server logic can sit.
Added an APP folder where the app logic can sit.
Separated out the different API areas from the one Routes file and into their own logical folders.
Moved passport operations into their own module.

### CODE REVIEW

File: expressjs/app.js [hunk 1]
```diff
@@ -1,64 +1,5 @@
...
-var expressSession = require('express-session');
-var bodyParser = require('body-parser');
-var cookieParser = require('cookie-parser');
-var path = require('path');
-const passport = require('passport');
-var logger = require('morgan');
+require('./paths.js')
 
-var indexRouter = require('./routes/index');
+const { server } = require('src/')
```
[the remark this answers is not in the document]
Comment (sublet) [reply expressjs/app.js]: Noted! Just pushed in an update that handles removing app-paths and adding relative links.  Additionally I added eslint and some rules for us to abide by :)

**Comment by krishnadurai** [comment]
Thanks @sublet for this refactor, let me know when I review this once it is ready.

**Comment by will-weiss** [comment]
Echo @aiyan - let's add a linter to get end formatting discussions (which are a perennial waste of time) and then LGTM.

**Comment by sublet** [comment]
@will-weiss I did add a linter.

**Pull request merged**